vevestaX/vevesta.py

```diff
@@ -812,26 +856,23 @@ def commit(self, techniqueUsed, filename=None, message=None, version=None, proje
         token = self.__find_access_token()
         backend_url = 'https://api.matrixkanban.com/services-1.0-SNAPSHOT'
 
-        headers = {'Authorization': 'Bearer ' + token}
-        params = {'projectId': projectId}
-
-        # get git access token
-        response = requests.post(url=backend_url + '/GetGitToken', headers=headers, params=params)
-        data = response.json()
-        git_token = data['gitToken']
-        project_title = data['projectTitle']
-        if repoName is None:
-            repoName = project_title
-        if sourceBranch is None:
-            sourceBranch = techniqueUsed
-        if newBranch is None:
-            newBranch = sourceBranch
+        # push to git
         try:
+            git_token = self.__find_git_token(backend_url=backend_url, access_token=token)
+            if repoName is None:
+                project = self.__fetch_project(backend_url=backend_url, access_token=token, projectId=projectId)
+                project_title = project['title']
+                repoName = project_title
+
+            if sourceBranch is None:
+                sourceBranch = techniqueUsed
+            if newBranch is None:
+                newBranch = sourceBranch
```

Comment:
new branch will be techniqueUsed...when source branch is main, then it that case a new branch say XGboost will be created